File: Python/Program /Shirt.py
# Mengimpor kelas Product
from Product import Product

# Mendefinisikan kelas Shirt yang merupakan turunan dari kelas Product
class Shirt(Product):
    # Konstruktor untuk kelas Shirt
    def __init__(self):
        # Memanggil konstruktor dari kelas induk (Product)
        super().__init__()
        # Menambahkan atribut color dengan nilai awal string kosong
        self.color = ""
        # Menambahkan atribut sleeve_type dengan nilai awal string kosong
        self.sleeve_type = ""

    # Metode untuk mengatur nilai dari atribut color
    def set_color(self, color):
        self.color = color

    # Metode untuk mengatur nilai dari atribut sleeve_type
    def set_sleeve_type(self, sleeve_type):
        self.sleeve_type = sleeve_type

    # Metode untuk mendapatkan nilai dari atribut color
    def get_color(self):
        return self.color

    # Metode untuk mendapatkan nilai dari atribut sleeve_type
    def get_sleeve_type(self):
        return self.sleeve_type

File: CPP/Program/Shirt.cpp
#include "Clothing.cpp"

class Shirt : public Clothing { // Deklarasi kelas bernama Shirt yang merupakan subkelas dari kelas Product
private:
    string color; // Melakukan deklarasi variabel private string untuk warna
    string sleeve_type; // Melakukan deklarasi variabel private string untuk tipe lengan

public:
    Shirt() { // Deklarasi konstruktor untuk kelas Shirt
        this->color = ""; // Melakukan inisialisasi warna 
        this->sleeve_type = ""; // Melakukan inisialisasi tipe lengan 
    }

    void set_color(string color) { // Deklarasi metode untuk mengatur warna
        this->color = color; // Menetapkan warna dengan nilai yang diberikan
    }

    void set_sleeve_type(string sleeve_type) { // Deklarasi metode untuk mengatur tipe lengan
        this->sleeve_type = sleeve_type; // Menetapkan tipe lengan dengan nilai yang diberikan
    }

    string get_color() { // Deklarasi metode untuk mendapatkan warna
        return this->color; // Mengembalikan warna
    }

    string get_sleeve_type() { // Deklarasi metode untuk mendapatkan tipe lengan
        return this->sleeve_type; // Mengembalikan tipe lengan
    }
};
